src/clr-angular/forms/datepicker/providers/date-io.service.spec.ts

```diff
@@ -54,135 +66,76 @@ export default function() {
-                dateIOService.inputDate = "01/02/2015";
-                expect(dateIOService.inputDate).toBe("01/02/2015");
-
-                dateIOService.inputDate = "01/02/201";
-                expect(dateIOService.inputDate).toBe("");
-            });
-
-            it("supports a date property which is populated according to the dateInput", () => {
-                expect(dateIOService.date).toBeUndefined();
-
-                dateIOService.inputDate = "01/02/2015";
-                expect(dateIOService.date).not.toBeUndefined();
-                expect(dateIOService.date).not.toBeNull();
-                expect(assertEqualDates(dateIOService.date, new Date(2015, 0, 2))).toBe(true);
-
-                // Leap Year
-                dateIOService.inputDate = "02/29/2016";
-                expect(assertEqualDates(dateIOService.date, new Date(2016, 1, 29))).toBe(true);
+            it("ignores text", () => {
+                const inputDate: string = "abc";
+                const date: Date = dateIOService.isValidInput(inputDate);
+                expect(date).toBeNull();
             });
 
-            it("sets the date property to null when an invalid date is entered", () => {
-                expect(dateIOService.date).toBeUndefined();
-                dateIOService.inputDate = "abcd";
-                expect(dateIOService.date).toBeUndefined();
-
-                dateIOService.inputDate = "01/02/2015";
-                expect(dateIOService.date).not.toBeUndefined();
-                expect(dateIOService.date).not.toBeNull();
-                expect(assertEqualDates(dateIOService.date, new Date(2015, 0, 2))).toBe(true);
+            it("ignores empty strings", () => {
+                const inputDate: string = "";
+                const date: Date = dateIOService.isValidInput(inputDate);
+                expect(date).toBeNull();
             });
 
             it("parse a two digit year", () => {
-                dateIOService.inputDate = "01/02/20";
-                expect(dateIOService.date).not.toBeNull();
-                expect(assertEqualDates(dateIOService.date, new Date(2020, 0, 2))).toBe(true);
+                let inputDate: string = "01/02/20";
+                let date: Date = dateIOService.isValidInput(inputDate);
+                expect(date).not.toBeNull();
+                expect(assertEqualDates(date, new Date(2020, 0, 2))).toBe(true);
 
                 // Invalid date with 2 digit year
-                dateIOService.inputDate = "51/02/20";
-                expect(dateIOService.date).toBeNull();
+                inputDate = "51/02/20";
+                date = dateIOService.isValidInput(inputDate);
+                expect(date).toBeNull();
             });
 
             it("should not parse a five digit year", () => {
-                dateIOService.inputDate = "01/02/10000";
-                expect(dateIOService.date).toBeUndefined();
+                const inputDate: string = "01/02/10000";
```

Comment:
Just a note, typing is overkill here as it _has_ to be a string and cannot change (`const`) so shouldn't be necessary. Nothing wrong with it just extra.